src/plugins/gui/providers/paybisProvider.ts

```diff
@@ -0,0 +1,470 @@
+          amount,
+          currencyCodeTo,
+          directionChange,
+          isReceivedAmount: directionChange === 'to',
+          paymentMethod
+        }
+        const response = await paybisFetch({ method: 'POST', url, path: 'v2/quote', apiKey, bodyParams })
+        const { id: quoteId, paymentMethods, paymentMethodErrors } = asQuote(response)
+
+        if (paymentMethodErrors != null) {
+          let lastError
+          for (const e of paymentMethodErrors) {
+            lastError = e
+            const maxMatch = e.error.message.match(/^Amount must be less than (\d+\.\d+) ([A-Z]+)/)
+            const minMatch = e.error.message.match(/^Minimum amount is (\d+\.\d+) ([A-Z]+)/)
+            if (maxMatch != null) {
+              throw new FiatProviderError({ providerId, errorType: 'overLimit', errorAmount: Number(maxMatch[1]), displayCurrencyCode: maxMatch[2] })
+            } else if (minMatch != null) {
+              throw new FiatProviderError({ providerId, errorType: 'underLimit', errorAmount: Number(minMatch[1]), displayCurrencyCode: minMatch[2] })
+            }
+          }
+          throw new Error(lastError?.error.message ?? 'Paybis Unknown paymentMethodError')
+        }
+
+        // Should only have one payment method
+        if (paymentMethods.length !== 1) {
+          throw new Error('Invalid number of quoted payment methods')
+        }
+
+        const pmQuote = paymentMethods[0]
+        const { id: paymentMethodId, amountFrom, amountTo } = pmQuote
+
+        let cryptoAmount: string
+        let fiatAmount: string
+
+        if (direction === 'buy') {
+          fiatAmount = amountFrom.amount
+          cryptoAmount = amountTo.amount
+          if (directionChange === 'from') {
+            // Sanity check the quote
+            assert(eq(amount, amountFrom.amount), 'Quote not equal to requested from amount')
+          } else {
+            assert(eq(amount, amountTo.amount), 'Quote not equal to requested to amount')
+          }
+        } else {
+          // XXX Todo
+          throw new Error('Sell not supported')
+        }
+
+        return {
+          providerId,
+          partnerIcon: 'dummy.png',
+          pluginDisplayName: 'Paybis',
+          displayCurrencyCode,
+          cryptoAmount,
+          isEstimate: false,
+          fiatCurrencyCode,
+          fiatAmount,
+          direction,
+          regionCode,
+          paymentTypes,
+          approveQuote: async (approveParams: FiatProviderApproveQuoteParams): Promise<void> => {
+            const { coreWallet, showUi } = approveParams
+            const receiveAddress = await coreWallet.getReceiveAddress()
+
+            const bodyParams = {
+              cryptoWalletAddress: {
+                currencyCode: paybisCc,
+                address: receiveAddress.segwitAddress ?? receiveAddress.publicAddress
+              },
+              partnerUserId,
+              locale: locale.localeIdentifier.slice(0, 2),
```

Comment:
We should start referencing the same locale logic i.e. pickLanguage() in
preparation for locale UI settings